Source/WebCore/rendering/RenderView.cpp

```diff
@@ -957,11 +957,13 @@ RenderLayer* RenderView::takeStyleChangeLayerTreeMutationRoot()
 void RenderView::registerBoxWithScrollSnapPositions(const RenderBox& box)
 {
     m_boxesWithScrollSnapPositions.add(&box);
+    m_boxesWithScrollSnapPositionsIDMap.add(&box, m_boxesWithScrollSnapPositionsCurrentID++);
```

Comment:
I wonder about this approach. The specification says that resnapping after layout should be preserved according the *element*, but IIUC style change can destroy and recreate an element's `RenderView`. Is there a way to rework this so that these identifiers are preserved when the `RenderView` is recreated?

Reply:
While I'm not too familiar with the lifetime of RenderView, at least while testing this change I didn't see the RenderView being destroyed due to any style changes. However, another solution is to use the `identifier()` function on each `Element`. This also allows us to not maintain a map anymore, which Simon had some concerns about previously. This also might prevent any lifetime issues.

Reply:
Thanks. The new approach looks great to me.